go/types/list.go

```diff
@@ -162,24 +162,30 @@ func (l List) RemoveAt(idx uint64) List {
 type listIterFunc func(v Value, index uint64) (stop bool)
 
 func (l List) Iter(f listIterFunc) {
+	l.IterFrom(0, f)
+}
+
+type listIterAllFunc func(v Value, index uint64)
+
+func (l List) IterAll(f listIterAllFunc) {
 	idx := uint64(0)
 	cur := newCursorAtIndex(l.seq, idx)
 	cur.iter(func(v interface{}) bool {
-		if f(v.(Value), uint64(idx)) {
-			return true
-		}
+		f(v.(Value), uint64(idx))
 		idx++
 		return false
 	})
 }
 
-type listIterAllFunc func(v Value, index uint64)
-
-func (l List) IterAll(f listIterAllFunc) {
-	idx := uint64(0)
+func (l List) IterFrom(idx uint64, f listIterFunc) {
```

Comment:
I'm wondering if we should break with tradition and make IterFrom a channel based API? It's more flexible, and you've gone through some contortions to adapt the callback API into a channel one.

(using the same idiom as diff where it takes the result and close channels, and sends back on the same goroutine)